src/Internal/Payment/State/InitialState.php

```diff
@@ -187,12 +194,34 @@ protected function populate_context_from_order() {
 			)
 		);
 		$context->set_level3_data( $this->level3_service->get_data_from_order( $order_id ) );
+	}
 
-		// Customer management involves a remote call.
-		$customer_id = $this->customer_service->get_or_create_customer_id_from_order(
-			$context->get_user_id(),
-			$this->order_service->_deprecated_get_order( $order_id )
-		);
+	/**
+	 * Parse and manage customer details.
+	 * This includes the update/creation of a customer and link it to the order.
+	 *
+	 * @throws Order_Not_Found_Exception In case the order could not be found.
+	 * @throws API_Exception
+	 * @throws Exception
```

Comment:
The API exception makes sense, and we can handle it together with intent-related API exceptions in the future, but where does this one come from? I don't see any methods in here that throw it. Let's see where it originates, and decide how to handle it, as states should not throw plain `Exception`s.

Reply:
This plan Exception is no longer thrown here since this part of code is moved into the customer service. The `new WC_Customer( $user_id ) `can throw plan `Exception`,  so I had to add it.